devicetypes/smartthings/zigbee-window-shade.src/zigbee-window-shade.groovy

```diff
@@ -90,33 +90,66 @@ def parse(String description) {
             List<Map> descMaps = collectAttributes(descMap)
             def liftmap = descMaps.find { it.attrInt == ATTRIBUTE_POSITION_LIFT }
             if (liftmap) {
-                state.level = Integer.parseInt(liftmap.value, 16)
                 if (liftmap.value == "64") { //open
-                    sendEvent(name: "windowShade", value: "open")
                     sendEvent(name: "level", value: "100")
+                    levelEventHandler(100)
                 } else if (liftmap.value == "00") { //closed
-                    sendEvent(name: "windowShade", value: "closed")
                     sendEvent(name: "level", value: "0")
+                    levelEventHandler(0)
                 } else {
-                    sendEvent(name: "windowShade", value: "partially open")
                     sendEvent(name: "level", value: zigbee.convertHexToInt(liftmap.value))
+                    levelEventHandler(zigbee.convertHexToInt(liftmap.value))
                 }
+                state.level = Integer.parseInt(liftmap.value, 16)
             }
         } else if (!supportsLiftPercentage() && descMap?.clusterInt == zigbee.LEVEL_CONTROL_CLUSTER && descMap.value) {
             def valueInt = Math.round((zigbee.convertHexToInt(descMap.value)) / 255 * 100)
-            state.level = valueInt
-            if (0 == valueInt) {
-                sendEvent(name: "windowShade", value: "closed")
-            } else if (100 == valueInt) {
-                sendEvent(name: "windowShade", value: "open")
-            } else {
-                sendEvent(name: "windowShade", value: "partially open")
-            }
+
+            levelEventHandler(valueInt)
             sendEvent([name: "level", value: valueInt])
+            state.level = valueInt
         }
     }
 }
 
+def levelEventHandler(currentLevel) {
+    log.debug "levelEventHandler:- ${currentLevel}"
+    def lastLevel = state.level
+    if (currentLevel == lastLevel) { //Ignore invalid reports
```

Comment:
This should probably be `if (lastLevel && currentLevel == lastLevel) {` in case lastLevel is null.

Reply:
ok, I fixed it.